app/components/NavMenu/index.tsx

```diff
@@ -96,12 +97,32 @@ interface Props {
 export const NavMenu: FC<Props> = (props) => {
   const locale = useSelector(selectLocale);
   const balances = useSelector(selectBalances);
-  const { pathname } = useLocation();
+  const domains = useSelector(selectDomain);
+  const [domain, setDomain] = useState<Domain | undefined>(undefined);
+  const svg = useRef(null);
+  useEffect(() => {
+    if (domains?.knsDomain.name) {
+      setDomain(domains.knsDomain);
+    } else if (domains?.ensDomain.name) {
+      setDomain(domains.ensDomain);
+    }
+  }, [domains]);
```

Comment:
no need to set to a secondary state `domain`, the useSelector hook should already trigger a rerender whenever `state.user.nameServiceDomains` changes

Reply:
Ok, I made the address section its own functional component to clean this up.